firefox-ios/Client/Frontend/Browser/Toolbars/Redux/AddressBarState.swift

```diff
@@ -124,449 +124,528 @@ struct AddressBarState: StateType, Equatable {
+            browserActions: state.browserActions,
+            borderPosition: state.borderPosition,
+            url: state.url,
+            searchTerm: state.searchTerm,
+            lockIconImageName: state.lockIconImageName,
+            safeListedURLImageName: state.safeListedURLImageName,
+            isEditing: state.isEditing,
+            isScrollingDuringEdit: state.isScrollingDuringEdit,
+            shouldSelectSearchTerm: state.shouldSelectSearchTerm,
+            isLoading: state.isLoading,
+            readerModeState: state.readerModeState,
+            didStartTyping: true,
+            showQRPageAction: true
+        )
+    }
+
+    private static func handleDidEnterSearchTermAction(state: Self, action: Action) -> Self {
+        guard let toolbarAction = action as? ToolbarAction else { return state }
+
+        return AddressBarState(
+            windowUUID: state.windowUUID,
+            navigationActions: state.navigationActions,
+            pageActions: pageActions(action: toolbarAction,
+                                     addressBarState: state,
+                                     isEditing: state.isEditing,
+                                     showQRPageAction: false),
+            browserActions: state.browserActions,
+            borderPosition: state.borderPosition,
+            url: state.url,
+            searchTerm: state.searchTerm,
+            lockIconImageName: state.lockIconImageName,
+            safeListedURLImageName: state.safeListedURLImageName,
+            isEditing: state.isEditing,
+            isScrollingDuringEdit: state.isScrollingDuringEdit,
+            shouldSelectSearchTerm: state.shouldSelectSearchTerm,
+            isLoading: state.isLoading,
+            readerModeState: state.readerModeState,
+            didStartTyping: true,
+            showQRPageAction: false
+        )
+    }
+
+    private static func handleDidStartTypingAction(state: Self, action: Action) -> Self {
+        guard action is ToolbarAction else { return state }
+
+        return AddressBarState(
+            windowUUID: state.windowUUID,
+            navigationActions: state.navigationActions,
+            pageActions: state.pageActions,
+            browserActions: state.browserActions,
+            borderPosition: state.borderPosition,
+            url: state.url,
+            searchTerm: state.searchTerm,
+            lockIconImageName: state.lockIconImageName,
+            safeListedURLImageName: state.safeListedURLImageName,
+            isEditing: state.isEditing,
+            isScrollingDuringEdit: state.isScrollingDuringEdit,
+            shouldSelectSearchTerm: state.shouldSelectSearchTerm,
+            isLoading: state.isLoading,
+            readerModeState: state.readerModeState,
+            didStartTyping: true,
+            showQRPageAction: state.showQRPageAction
+        )
+    }
+
+    private static func handleDefaultAction(state: Self) -> Self {
+        return AddressBarState(
```

Comment:
I think here @ionixjunior  we should clear all the transient state and use the default properties in the init.
```swift
return AddressBarState(
         windowUUID: state.windowUUID,
         navigationActions: state.navigationActions,
         pageActions: state.pageActions,
         browserActions: state.browserActions,
         borderPosition: state.borderPosition,
         url: state.url)
```

Would do you think @OrlaM ? is there a reason why we are passing all the previous state properties to the default action state ?

Reply:
The default is called when none of the other switch cases match. In this case we have to return a new struct with the current state as the default will be called all of the time (e.g. when a tab tray action is handled, the address toolbar needs to keep its current state).